test/unit/data.js

```diff
@@ -998,3 +998,18 @@ QUnit.test( ".data(prop) does not create expando", function( assert ) {
 		}
 	}
 } );
+
+QUnit.test( "keys matching Object.prototype properties  (gh-3256)", function( assert ) {
+	assert.expect( 2 );
+
+	var div = jQuery( "<div/>" );
+
+	assert.strictEqual( div.data( "hasOwnProperty" ), undefined,
+		"hasOwnProperty not matched (before forced data creation)" );
+
+	// Query the collection to force the creation of a data object for this element.
+	div.data();
```

Comment:
I think this should be even stronger.
```suggestion
	div.data( { foo: "bar" } );
```

Reply:
Good point, updated.